### ISSUE TITLE
How does GSO work in the project

### ISSUE THREAD

**Issue by Ellie-fans** [facebookincubator/mvfst#141]
Hi~I have looked up some blog about GSO, when I back to the project, I encount some problem
1. what are the differences between BATCHING_MODE_GSO, BATCHING_MODE_SENDMMSG and BATCHING_MODE_SENDMMSG_GSO?
2. I am trying to turn off/on gso using Mellanox and Inter NIC, testing download in server or uploading in client, the performance sometimes better but sometimes even worse. I use iftop to show real time traffic.

**Comment by mjoras** [comment]
TL;DR use BATCHING_MODE_GSO and forget about it :) 

The differences are:

BATCHING_MODE_GSO - uses one sendmsg call per write with GSO. Each datagram has to be equal in size except the last one.
BATCHING_MODE_SENDMMSG - uses one sendmmsg call per write. Datagrams can be any size.
BATCHING_MODE_SENDMMSG_GSO - combines the two above, one sendmmsg per write, each msg in the sendmmsg being submitted with GSO. Same rules per sendmmsg, but can save a syscall.

Theoretically the third one has the least syscall overhead, but for most cases it may not be measurable. Either of the second two should work transparently with hardware GSO.

**Comment by Ellie-fans** [comment]
Thanks a lot!
